c++/src/capnp/compiler/module-loader.c++

```diff
@@ -272,6 +275,14 @@ kj::Maybe<kj::Array<const byte>> ModuleLoader::Impl::readEmbedFromSearchPath(kj:
   return nullptr;
 }
 
+void ModuleLoader::Impl::reportResolution(uint32_t startByte, uint32_t endByte, uint64_t typeId) {
+  auto identifier = kj::heap<capnp::MallocMessageBuilder>().get()->initRoot<schema::CodeGeneratorRequest::RequestedFile::FileSourceInfo::Identifier>();
```

Comment:
A `Reader` is like a pointer. If the message it is pointing into is destroyed, the `Reader` is no longer valid. Here, the `MassogMessageBuilder` will be destroyed at the end of the statement, since it's not being stored anywhere, so `identifier` is left dangling.

Per my other comment we should probably just store C++ structs instead of capnp structs.

Reply:
Thank you for your suggestion. I apologize for my lack of knowledge. It has been corrected.